HuntTheWumpus/HuntTheWumpus.cpp

```diff
@@ -201,66 +200,90 @@ int main()
+
+            if (stringTokens.size() == 0)
             {
-                std::cout << "A Move command must be followed by the destination cave id.\n";
-                continue;
+                throw std::runtime_error("");
             }
 
-            // Second token is a destination.
-            const auto destCave = std::stoi(stringTokens[1]);
-
-            dungeon.MakeMove(HuntTheWumpus::DungeonMove::Move, { destCave });
-        }
+            const auto command = ToLower(stringTokens[0]);
 
-        if (command[0] == 's')
-        {
-            if (stringTokens.size() < 2)
+            if (command[0] == 'm' && command.length() != 0)
             {
-                std::cout << "A Shoot command must be followed by a list of caves for the arrow to go through.\n";
-                continue;
-            }
+                if (stringTokens.size() < 2)
+                {
+                    std::cout << "A Move command must be followed by the destination cave id.\n";
+                    continue;
+                }
+
 
-            // Remaining tokens is the desired arrow path.
-            std::vector<int> path;
-            auto firstToken = false;
+                // Second token is a destination.
+                const auto destCave = std::stoi(stringTokens[1]);
 
-            for (auto&& token : stringTokens)
+                dungeon.MakeMove(HuntTheWumpus::DungeonMove::Move, { destCave });
+            }
+            else if (command[0] == 's')
             {
-                if (!firstToken)
+                if (stringTokens.size() < 2)
                 {
-                    firstToken = true;
+                    std::cout << "A Shoot command must be followed by a list of caves for the arrow to go through.\n";
                     continue;
                 }
 
-                path.push_back(std::stoi(token));
-            }
+                // Remaining tokens is the desired arrow path.
+                std::vector<int> path;
+                auto firstToken = false;
 
-            path.resize(std::min(path.size(), PathLimit));
+                for (auto&& token : stringTokens)
+                {
+                    if (!firstToken)
+                    {
+                        firstToken = true;
+                        continue;
+                    }
 
-            dungeon.MakeMove(HuntTheWumpus::DungeonMove::Shoot, path);
-        }
+                    path.push_back(std::stoi(token));
+                }
+
+                path.resize(std::min(path.size(), PathLimit));
 
-        if (command[0] == 'q' || command[0] == 'e' || command[0] == 'x')
+                dungeon.MakeMove(HuntTheWumpus::DungeonMove::Shoot, path);
+            }
+            else if (command[0] == 'q' || command[0] == 'e' || command[0] == 'x')
+            {
+                std::cout << "Exiting.\n";
+                change.GameOver(false);
+            }
+            else
+            {
+                throw std::invalid_argument("");
```

Comment:
Consider the control flow here. You're throwing, which is a flow-control to the catch block, which is effectively the next line.

I suspect you can have the logic you've got for std::invalid_argument in line instead of in a catch block.